File: fc-chat/demo/part3/ch3/05-1-message-system-add-channel/src/main/java/net/prostars/messagesystem/handler/websocket/KeepAliveHandler.java
package net.prostars.messagesystem.handler.websocket;

import net.prostars.messagesystem.constant.IdKey;
import net.prostars.messagesystem.dto.domain.UserId;
import net.prostars.messagesystem.dto.websocket.inbound.KeepAlive;
import net.prostars.messagesystem.service.SessionService;
import org.springframework.stereotype.Component;
import org.springframework.web.socket.WebSocketSession;

@Component
@SuppressWarnings("unused")
public class KeepAliveHandler implements BaseRequestHandler<KeepAlive> {

  private final SessionService sessionService;

  public KeepAliveHandler(SessionService sessionService) {
    this.sessionService = sessionService;
  }

  @Override
  public void handleRequest(WebSocketSession senderSession, KeepAlive request) {
    UserId senderUserId = (UserId) senderSession.getAttributes().get(IdKey.USER_ID.getValue());

    sessionService.refreshTTL(
        senderUserId, (String) senderSession.getAttributes().get(IdKey.HTTP_SESSION_ID.getValue()));
  }
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/handler/websocket/KeepAliveHandler.kt
package com.yscorp.withpush.messagesystem.handler.websocket

import com.yscorp.withpush.messagesystem.constant.IdKey
import com.yscorp.withpush.messagesystem.dto.domain.UserId
import com.yscorp.withpush.messagesystem.dto.websocket.inbound.KeepAlive
import com.yscorp.withpush.messagesystem.service.SessionService
import net.prostars.messagesystem.constant.IdKey
import org.springframework.stereotype.Component
import org.springframework.web.socket.WebSocketSession

@Component
@Suppress("unused")
class KeepAliveHandler(
    private val sessionService: SessionService
) : BaseRequestHandler<KeepAlive> {

    /**
     * 클라이언트의 연결 유지를 위한 KeepAlive 요청을 처리합니다.
     * 해당 세션의 TTL(Time To Live)을 연장합니다.
     */
    override fun handleRequest(senderSession: WebSocketSession, request: KeepAlive) {
        val senderUserId = senderSession.attributes[IdKey.USER_ID.value] as? UserId ?: return
        val httpSessionId = senderSession.attributes[IdKey.HTTP_SESSION_ID.value] as? String ?: return

        sessionService.refreshTTL(senderUserId, httpSessionId)
    }
}
